src/js/d3Utils.js

```diff
@@ -23,6 +23,42 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         return d3.selectAll(elem.toArray());
     };
 
+    /* Given the object "d", returns value of:
+     * - "id" if it exists on the top-level object,
+     * - "data.id" if it exists on a "data" object
+     * - "undefined" if neither of these exist
+     */
+    floe.d3.idExtractor = function (d) {
+        try {
+            var id = d.id !== undefined ? d.id : d.data.id;
+            return id !== undefined ? id : undefined;
+        } catch (e) {
+            return undefined;
+        }
+    };
+
+    // Given a selection of D3 elements and an ID, returns only the elements
+    // matching that ID
+    floe.d3.filterById = function(d3Selection, currentlyPlayingDataId) {
```

Comment:
Not QUITE identical, I see - now I've looked at the above implementation in more detail, I see that these utilities are involved as part of the general strategy of "using our view as a model" which should be factored away. We should track the relationships between playing data, rows and slice ids at the model level of the application rather than storing it in the DOM.